### ISSUE TITLE
On-call Support for Haqdarshak Team to Execute Beckn-Onix

### ISSUE THREAD

**Issue by emmayank** [beckn/beckn-utilities#136]
## Description
Provide on-call support to the Haqdarshak team to assist them in executing the beckn-onix project. This involves offering real-time assistance, troubleshooting issues, and ensuring smooth execution of the project. 

## Goals
- [x] Establish a support schedule with the Haqdarshak team.
- [x] Provide real-time assistance during the execution of beckn-onix.
- [x] Troubleshoot and resolve any issues that arise.
- [x] Ensure the project is executed smoothly and efficiently.

## Expected Outcome
The Haqdarshak team successfully executes the beckn-onix project with minimal disruptions, thanks to the on-call support provided. All issues are resolved promptly, and the project progresses smoothly.

## Acceptance Criteria
- [x] A support schedule is established and communicated to the Haqdarshak team.
- [x] On-call support is provided as per the schedule.
- [x] All issues encountered during the execution are resolved promptly.
- [x] The project is executed without significant delays or disruptions.

## Mockups / Wireframes
NA

## Product Name
Beckn

## Domain
Beckn-Onix

## Tech Skills Needed
- Strapi
- Node.js
- JavaScript
- Troubleshooting
- Real-time support

## Complexity
Medium

## Category
Support

## Sub Category
On-call Support

**Comment by prasad-takale-eminds** [comment]
The Haqdarshak team encountered an error while installing Beckn-ONIX on their Debian machine:
```
................Installing Registry service................
unknown shorthand flag: 'f' in -f
See 'docker --help'.
```

This error occurred because the if we are using older docker version on the system. Also `docker-compose` is no longer actively maintained ([Link](https://docs.docker.com/reference/compose-file/legacy-versions/)). We have removed `docker-compose` from the Beckn-ONIX installation script and replaced it with the `docker compose` command.

To resolve this issue, we updated Docker to version 27.1.2 on their Debian machine and re-ran the installation, which was completed successfully.

**Comment by faizmagic** [comment]
@emmayank , Please take the necessary stpes to ensure that the corresponding documentation and software is fixed, so that our team does not need to engage for similar support.

CC: @prasad-takale-eminds , @viraj89

**Comment by prasad-takale-eminds** [comment]
@faizmagic,  We have added support to the docker installation for Debian in our script.  PR for the same is [here](https://github.com/beckn/beckn-onix/pull/190).

To resolve the Haqdarshak Beckn-ONIX installation issue, we followed these steps:

1. **Identified the Issue**: We first identified the problem that was causing the installation to fail.
   
2. **Debugged the ONIX Script**: We checked our ONIX script to determine where the issue originated.
   
3. **Removed the Old Docker Version**: We attempted to remove the outdated Docker version from their machine. This took more time than expected as the `apt remove docker` command did not fully uninstall Docker.
   
4. **Installed the Correct Docker Version**: We then installed the correct version of Docker on their machine. This required finding the appropriate Docker repository for Debian.

5. **Reinstalled Beckn-ONIX**: After updating Docker, we reinstalled Beckn-ONIX. During this step, we encountered a new issue where the Network Participant (NP) was already registered on their registry. We removed the NP entry and proceeded with the installation.

6. **Final Installation and Troubleshooting**: The installation was successful, but the team couldn’t access the protocol server using the URL, despite the application running without errors. We identified that the issue was on their end and guided their team on how to resolve it. After following our instructions, the URL became accessible.
